src/update-feeds/check-feeds.test.ts

```diff
@@ -0,0 +1,265 @@
+import { ethers } from 'ethers';
+
+import { initializeState } from '../../test/fixtures/mock-config';
+import { allowPartial } from '../../test/utils';
+import * as signedDataStore from '../signed-data-store/signed-data-store';
+import { updateState } from '../state';
+import type { DataFeedId, SignedData } from '../types';
+
+import * as contractUtils from './api3-server-v1';
+import { callAndParseMulticall, deeplyCheckDapis, shallowCheckFeeds } from './check-feeds';
+import * as checkFeeds from './check-feeds';
+import type { ReadDapiWithIndexResponse } from './dapi-data-registry';
+import { encodeBeaconValue } from './update-feeds';
+
+describe('feed update checks', () => {
+  describe('shallow checks', () => {
+    it('returns a feed as updatable', () => {
+      initializeState();
+      updateState((draft) => {
+        draft.signedApiStore = allowPartial<Record<DataFeedId, SignedData>>({
+          '0x000a': { timestamp: '100', encodedValue: encodeBeaconValue('200') },
+          '0x000b': { timestamp: '150', encodedValue: encodeBeaconValue('250') },
+          '0x000c': { timestamp: '200', encodedValue: encodeBeaconValue('300') },
+        });
+      });
+
+      const batch = allowPartial<ReadDapiWithIndexResponse[]>([
+        {
+          updateParameters: { deviationThresholdInPercentage: ethers.BigNumber.from(2) },
+          dataFeedValue: { value: ethers.BigNumber.from(1), timestamp: 1 },
+          decodedDataFeed: {
+            dataFeedId: '0x000',
+            beacons: [{ beaconId: '0x000a' }, { beaconId: '0x000b' }, { beaconId: '0x000c' }],
+          },
+          dapiName: 'test',
+        },
+      ]);
+      const shallowCheckResult = shallowCheckFeeds(batch, 2);
+
+      expect(shallowCheckResult).toHaveLength(1);
+    });
+
+    it('returns an empty array for non-updatable feed', () => {
+      initializeState();
+      updateState((draft) => {
+        draft.signedApiStore = allowPartial<Record<DataFeedId, SignedData>>({
+          '0x000a': { timestamp: '100', encodedValue: encodeBeaconValue('200') },
+          '0x000b': { timestamp: '150', encodedValue: encodeBeaconValue('250') },
+          '0x000c': { timestamp: '200', encodedValue: encodeBeaconValue('300') },
+        });
+      });
+
+      const batch = allowPartial<ReadDapiWithIndexResponse[]>([
+        {
+          updateParameters: { deviationThresholdInPercentage: ethers.BigNumber.from(2) },
+          dataFeedValue: { value: ethers.BigNumber.from(150), timestamp: 150 },
```

Comment:
nit: It would be nice to set `timestamp: 300` so its obvious it fails because of the timestamp check.